src/no_std_error.rs

```diff
@@ -0,0 +1,45 @@
+#![allow(missing_docs)]
+
+use core::fmt::{Debug, Display};
+
+pub trait Error: Debug + Display {
+    fn description(&self) -> &str {
+        "description() is deprecated; use Display"
+    }
+    fn cause(&self) -> Option<&dyn Error> {
+        self.source()
+    }
+    fn source(&self) -> Option<&(dyn Error + 'static)> {
+        None
+    }
+}
+
+macro_rules! impl_error {
+    ($($e:path),*) => {
+        $(
+            impl Error for $e {}
+        )*
+    }
+}
+
+// All errors supported by our minimum suported Rust version can be supported by
+// default.
+impl_error![
+    core::str::ParseBoolError,    // 1.0
+    core::str::Utf8Error,         // 1.0
+    core::num::ParseIntError,     // 1.0
+    core::num::ParseFloatError,   // 1.0
+    core::char::DecodeUtf16Error, // 1.9
+    core::fmt::Error,             // 1.11
+    core::cell::BorrowMutError,   // 1.13
+    core::cell::BorrowError,      // 1.13
+    core::char::ParseCharError    // 1.20
```

Comment:
Since we bumped our minimum version to 1.31, are there any errors missing from this list?

Reply:
No that I'm aware of. When I created the PR these were all the Error impls in std. These used to be three macro calls where one had the rust_1_30 feature check, but I condensed that into the unconditional macro call, so I don't think I missed any (other than potential ones in 1.38+ or so).